agent/client.go

```diff
@@ -38,6 +43,95 @@ type client struct {
 	cr               CredentialsReceiver
 }
 
+type accessKeyClient struct {
+	sts         *sts.STS
+	iamAccount  string
+	iamUsername string
+	cr          CredentialsReceiver
+}
+
+func AccessKeyClient(accessKey string, secretKey string, cr CredentialsReceiver) *accessKeyClient {
+	sts := sts.New(session.New(&aws.Config{Credentials: credentials.NewStaticCredentials(accessKey, secretKey, "")}))
+	iamconn := iam.New(session.New(&aws.Config{Credentials: credentials.NewStaticCredentials(accessKey, secretKey, "")}))
+	iamUser, err := iamconn.GetUser(&iam.GetUserInput{})
+	if err != nil {
+		log.Errorf("Unable to get current user.  Err: %s", err)
+	}
+	iamAccount := strings.Split(*iamUser.User.Arn, ":")[4]
+	iamUsername := iamUser.User.UserName
+
+	c := &accessKeyClient{
+		sts:         sts,
```

Comment:
yep, exactly.  these sts credentials cant call IAM or STS, or get access to the console. you'd have to assume a role first to do those things which is kinda lame